packages/camelot/src/Camelot.test.ts

```diff
@@ -0,0 +1,54 @@
+import { GreaterThanOrEqual, apply } from '@rabbitholegg/questdk/filter'
+import { describe, expect, test } from 'vitest'
+import { SWAP_ETH, SWAP_TOKENS } from './test-transactions'
+import { CAMELOT_ROUTER } from './contract-addresses'
+import { ARBITRUM_CHAIN_ID } from './chain-ids'
+import { parseEther } from 'viem'
+import { swap } from './Camelot'
+import { CAMELOT_ABI } from './abi'
+describe('Given the camelot plugin', () => {
+  describe('When handling the bridge', () => {
+    test('should return a valid action filter', async () => {
+      const filter = await swap({
+        chainId: ARBITRUM_CHAIN_ID,
+        contractAddress: CAMELOT_ROUTER,
+        tokenOut: '0xBfbCFe8873fE28Dfa25f1099282b088D52bbAD9C',
+        amountOut: GreaterThanOrEqual(parseEther('0.0005')),
+      })
+
+      expect(filter).to.deep.equal({
+        chainId: 42161,
+        to: '0xc873fEcbd354f5A56E00E710B90EF4201db2448d',
+        input: {
+          $abi: CAMELOT_ABI,
+          path: {
+            $and: [{ $last: '0xBfbCFe8873fE28Dfa25f1099282b088D52bbAD9C' }],
+          },
+          amountOutMin: { $gte: '500000000000000' },
+        },
+      })
+    })
+    test('should pass filter with valid ETH transactions', async () => {
+      const transaction = SWAP_ETH
+      const filter = await swap({
+        chainId: ARBITRUM_CHAIN_ID,
+        contractAddress: CAMELOT_ROUTER,
+        tokenOut: '0xBfbCFe8873fE28Dfa25f1099282b088D52bbAD9C',
+        amountOut: GreaterThanOrEqual(parseEther('0.0005')),
+      })
+      console.log(filter)
```

Comment:
Can remove this log

Reply:
I removed a few other hanging ones - I'm so bad at removing logs I'd make a terrible lumberjack